server/eclipse-project/src/main/java/us/freeandfair/corla/query/CastVoteRecordQueries.java

```diff
@@ -426,7 +427,31 @@ public static CastVoteRecord atPosition(final Tribute tribute) {
 
     results.addAll(phantomRecords);
 
-    return results;
+    // this is a dummy list so we can add a cvr at a particular position(that of
+    // the tributes uris)
+    final List<CastVoteRecord> randomOrder =
+      new ArrayList<CastVoteRecord>(Collections.nCopies(uris.size(), null));
+
+    // line the cvrs back up into the random order
+    for (final CastVoteRecord cvr: results) {
+      int index = 0;
+      for (final String uri: uris) {
+        if (uri.equals(cvr.getUri())) {
+          randomOrder.add(index, cvr);
+        }
+        index++;
+      }
```

Comment:
could

```java
int index = uris.indexOf(cvr.getUri());
if (-1 != index) {
  randomOrder.add(index, cvr);
}
```

work here?

Reply:
that's what I had initially but that doesn't handle duplicates